packages/react/src/hooks/participant-hooks.ts

```diff
@@ -14,38 +15,36 @@ import {
   mutedObserver,
   ParticipantMedia,
   observeParticipantMedia,
-  ParticipantFilter,
   participantPermissionObserver,
   connectedParticipantObserver,
+  UpdateOnlyOn,
+  toRoomEvents,
 } from '@livekit/components-core';
 import { useEnsureParticipant, useRoomContext } from '../context';
 import { useObservableState } from '../helper/useObservableState';
 
 export interface UseParticipantsOptions {
-  filter?: ParticipantFilter;
-  filterDependencies?: Array<unknown>;
+  updateOnlyOn?: UpdateOnlyOn;
 }
 
 /**
- * The useParticipants hook returns all participants of the current room.
- * It is possible to filter the participants.
+ * The useParticipants hook returns all participants (local and remote) of the current room.
+ *
+ * @remarks
+ * To optimize performance, you can use the `updateOnlyOn` property to take control of when the hook updates.
+ * By default the hook updates on all relevant RoomEvents to keep the returned participants array up to date.
  */
 export const useParticipants = (options: UseParticipantsOptions = {}) => {
   const [participants, setParticipants] = React.useState<Participant[]>([]);
-  const remoteParticipants = useRemoteParticipants({ filter: undefined });
+  const { updateOnlyOn } = options;
+  const remoteParticipants = useRemoteParticipants({
+    updateOnlyOn: updateOnlyOn ? toRoomEvents(updateOnlyOn) : [],
+  });
   const { localParticipant } = useLocalParticipant();
-  const filterDependencies = React.useMemo(
-    () => options?.filterDependencies ?? [],
-    [options?.filterDependencies],
-  );
 
   React.useEffect(() => {
```

Comment:
can we maybe even get rid of the useEffect here without causing constant re-renders?

Reply:
Lets try! 👍